apps/teca_cf_restripe.cpp

```diff
@@ -0,0 +1,320 @@
+                    << std::endl
+                    << all_opt_defs << std::endl
+                    << std::endl;
+                return -1;
+            }
+        }
+
+        boost::program_options::notify(opt_vals);
+    }
+    catch (std::exception &e)
+    {
+        TECA_ERROR("Error parsing command line options. See --help "
+            "for a list of supported options. " << e.what())
+        return -1;
+    }
+
+    // pass command line arguments into the pipeline objects
+    // advanced options are processed first, so that the basic
+    // options will override them
+    cf_reader->set_properties("cf_reader", opt_vals);
+    cf_writer->set_properties("cf_writer", opt_vals);
+
+    // now pass in the basic options, these are processed
+    // last so that they will take precedence
+    if (opt_vals.count("input_file"))
+        cf_reader->append_file_name(
+            opt_vals["input_file"].as<std::string>());
+
+    if (opt_vals.count("input_regex"))
+        cf_reader->set_files_regex(
+            opt_vals["input_regex"].as<std::string>());
+
+    if (opt_vals.count("output_file"))
+        cf_writer->set_file_name(
+            opt_vals["output_file"].as<std::string>());
+
+    if (opt_vals.count("point_arrays"))
+        cf_writer->set_point_arrays(
+            opt_vals["point_arrays"].as<std::vector<std::string>>());
+
+    if (opt_vals.count("information_arrays"))
+        cf_writer->set_information_arrays(
+            opt_vals["information_arrays"].as<std::vector<std::string>>());
+
+    if (opt_vals.count("steps_per_file"))
+        cf_writer->set_steps_per_file(
+            opt_vals["steps_per_file"].as<long>());
+
+    if (opt_vals.count("first_step"))
+        cf_writer->set_first_step(opt_vals["first_step"].as<long>());
+
+    if (opt_vals.count("last_step"))
+        cf_writer->set_last_step(opt_vals["last_step"].as<long>());
+
+    if (opt_vals.count("verbose"))
+    {
+        cf_writer->set_verbose(1);
+        exec->set_verbose(1);
+    }
+
+    if (opt_vals.count("n_threads"))
+        cf_writer->set_thread_pool_size(opt_vals["n_threads"].as<int>());
+    else
+        cf_writer->set_thread_pool_size(1);
+
+    // some minimal check for missing options
+    if (cf_reader->get_number_of_file_names() == 0
+        && cf_reader->get_files_regex().empty())
+    {
+        if (mpi_man.get_comm_rank() == 0)
+        {
+            TECA_ERROR(
+                "missing file name or regex for the NetCDF CF reader. "
+                "See --help for a list of command line options.")
+        }
+        return -1;
+    }
+
+    if (cf_writer->get_file_name().empty())
+    {
+        if (mpi_man.get_comm_rank() == 0)
+        {
+            TECA_ERROR(
+                "missing file name pattern for the NetCDF CF writer. "
+                "See --help for a list of command line options.")
+        }
+        return -1;
+    }
+
+    // if no point arrays were specified on the command line by default
```

Comment:
the comment on line 209 seems to suggest that the code from lines 211-246 should do something; but that code is commented out.  I'd suggest revising the comment on 209 to reflect what the currently uncommented code is doing, and it might make sense to remove the commented code if it isn't going to be used in this release.

Reply:
done